addons/binding/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/handler/ShadeHandler.java

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.lutron.handler;
+
+import static org.openhab.binding.lutron.LutronBindingConstants.CHANNEL_SHADELEVEL;
+
+import java.math.BigDecimal;
+
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.library.types.StopMoveType;
+import org.eclipse.smarthome.core.library.types.UpDownType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.lutron.internal.protocol.LutronCommandType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Handler responsible for communicating with a Lutron Sivoia QS shade
+ *
+ * @author Bob Adair - Initial contribution
+ *         Based on Alan Tong's DimmerHandler
+ */
+public class ShadeHandler extends LutronHandler {
+    private static final Integer ACTION_ZONELEVEL = 1;
+    private static final Integer ACTION_STARTRAISING = 2;
+    private static final Integer ACTION_STARTLOWERING = 3;
+    private static final Integer ACTION_STOP = 4;
+
+    private final Logger logger = LoggerFactory.getLogger(ShadeHandler.class);
+
+    protected int integrationId;
+
+    public ShadeHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public int getIntegrationId() {
+        return integrationId;
+    }
+
+    @Override
+    public void initialize() {
+        Number id = (Number) getThing().getConfiguration().get("integrationId");
+        if (id == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "No integrationId");
+            return;
+        }
+        integrationId = id.intValue();
+
+        logger.debug("Initializing Shade handler for integration ID {}", id);
+
+        if (getThing().getBridgeUID() == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "No bridge configured");
+            return;
+        }
+        updateStatus(ThingStatus.ONLINE);
```

Comment:
Well, that's an interesting question.  This is using the same logic as all of the other device handlers in the binding at the moment.   It generally works, because the Lutron main repeater (the bridge) essentially hides the status of the end-devices from the integration interface.  It will handle it if they are unreachable.  So it is safe to assume they are online if the bridge is online.  That said, there needs to be checking to see if the bridge is online, and better checking of the connection status to the repeater in IPBridgeHandler.  I think that should all be done in a separate PR , though. I'd rather just leave this consistent with the other device handlers for now.

Reply:
> This is using the same logic as all of the other device handlers in the binding at the moment.

I saw that and from my point of view it is wrong for all other handlers too.

> I think that should all be done in a separate PR , though.

Yes, there is no need to change it for all in this PR. We should focus on the `ShadeHandler`.

What will happen if the bridge is `ONLINE` and the request fails because the `integrationId` of this handler is wrong (e.g. a typo)? Is the handler still `ONLINE`?

Reply:
Ok. I pushed a commit that I think will resolve this.  If it looks reasonable, I'll work on a new PR to fix it across all of the other RA2 handlers.